docs/source/raft_configuration.md

```diff
@@ -330,5 +330,34 @@ channels (this does not mean they share data --- they do not!), if a Raft node i
 part of the consenter set in many channels, you might want to lengthen the amount
 of time it takes to trigger an election to avoid inadvertent leader elections.
 
+2. As aforementioned, write-ahead logs (wal) and snapshots should be persisted in
+order to attain safety property of Raft protocol. However, it may happen if a node
+loses its data by accident, and the node would panic with following message:
+```
+tocommit(X) is out of range [lastIndex(Y)]. Was the raft log corrupted, truncated, or lost?
+```
+Manual operation would be required to resurrect problematic node in this case.
+* Safe recovery:  
+The only way to handle the corruption or loss of a WAL is to remove the corrupted
+consenter from the quorum, and add a new one in its place, which can be done following
+instructions in reconfiguration section in this doc. This is the most appropriate
+option, as it does not violate safety property of Raft protocol. However, this is heavy
+operation and affects all channels this node participates in.
+* Unsafe recovery:  
+It is possible to recover from a WAL corruption without removing the consenter from the
+consenter set and re-adding it. However, these procedures violate the Raft protocol safety
+guarantees and, in certain unlikely cases, could result in a blockchain fork. To minimize
+these possibilities, ensure that your cluster has quorum on all channels for a few minutes
+before proceeding.
+  * Option 1: wipe out wal and snapshot directories of problematic channel from the corrupted
+node, restart current leader node, then start corrupted orderer. This essentially treats
+corrupted orderer as a fresh node without any data. This is the simplest option that requires
+restarting leader node, which often leads to system downtime.
+  * Option 2: to avoid downtime caused by restarting leader node, you could also wipe out
+wal and snapshot directories of problematic channel from the corrupted node, and copy
+a snapshot from an alive node with greater index than `X` in above panic message. The index
```

Comment:
Option 2: To avoid downtime caused by restarting the leader node, you could also delete the
wal and snapshot directories of the problematic channel from the corrupted node, and then copy
a snapshot from an alive node with an index greater than `X` in above panic message.

Reply:
Why are we documenting an unsafe practice we explicitly don't recommend?

Reply:
The unsafe options are significantly simpler and the risk is fairly low. We do not explicitly recommend one over another (and we do have some sentences saying how to lower the probability of chain fork). But we definitely should document the risk, to be complete.

Reply:
Why wouldn't we explicitly recommend the safe path? 

I still think telling people how to do something we're calling "unsafe" is a bad idea.

Reply:
That's definitely a valid concern. IMHO though, explicit "use with caution" is better than "hidden feature", especially when the safe option does introduce tremendous amount of overhead, and the unsafe option is actually not very unsafe. I think this is still a handy solution to most of users.

@jyellick what do you think?